Sources/Magnetic.swift

```diff
@@ -103,44 +123,90 @@ extension Magnetic {
     
     override open func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
         if let touch = touches.first {
-            let location = touch.location(in: self)
-            let previous = touch.previousLocation(in: self)
-            
-            if location.distance(from: previous) == 0 { return }
-            
-            isDragging = true
-            
-            let x = location.x - previous.x
-            let y = location.y - previous.y
-            
-            for node in children {
-                let distance = node.position.distance(from: location)
-                let acceleration: CGFloat = 3 * pow(distance, 1/2)
-                let direction = CGVector(dx: x * acceleration, dy: y * acceleration)
+            let touchLocation = touch.location(in: self)
+            if movingNode == nil &&
+                allowSingleNodeMovement &&
+                initialTouchLocation == nil{
+                for node in children {
+                    let nodeTouchPoint=node.convert(touchLocation, to: node)
+                    if node.frame.contains(nodeTouchPoint)
+                    {
+                        movingNode = node
+                        if initialTouchLocation == nil{
+                            initialTouchStartedOnNode = true
+                        }
+                        break
+                    }
+                }
+            }
+            if !isDragging{
+                isDragging = true
+                initialTouchLocation = touchLocation
+            }
+            if allowSingleNodeMovement && initialTouchStartedOnNode, let node = movingNode{
+                let convertedTapLocation = convert(touchLocation, to: node)
+                let direction = CGVector(dx: convertedTapLocation.x * 30, dy: convertedTapLocation.y * 30)
                 node.physicsBody?.applyForce(direction)
+                
+                if movingNodeTimer != nil{
+                    movingNodeTimer?.invalidate()
+                    movingNodeTimer = nil
+                }
+                movingNodeTimer = Timer.scheduledTimer(timeInterval: 0.01, target: self, selector: #selector(self.moveNode(timer:)), userInfo: ["touchLocation":touchLocation, "node": node], repeats: true)
```

Comment:
use a Timer block instead of a selector
https://developer.apple.com/documentation/foundation/timer/2091889-scheduledtimer